stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/OnrampInteractor.kt

```diff
@@ -242,6 +251,132 @@ internal class OnrampInteractor @Inject constructor(
     fun onLinkControllerState(linkState: LinkController.State) {
         _state.value = _state.value.copy(linkControllerState = linkState)
     }
+
+    /**
+     * Starts the checkout flow for a crypto onramp session. The checkout state will be emitted
+     * through the state StateFlow. The coordinator should observe the checkoutState and react accordingly.
+     *
+     * @param onrampSessionId The onramp session identifier.
+     * @param checkoutHandler An async closure that calls your backend to perform a checkout.
+     */
+    suspend fun startCheckout(
+        onrampSessionId: String,
+        checkoutHandler: suspend () -> String
+    ) {
+        // Start processing with session info
+        _checkoutState.update {
+            it.copy(
+                onrampSessionId = onrampSessionId,
```

Comment:
follow-up: I think we should more explicitly handle what happens if `startCheckout` is called multiple times. Maybe do nothing if `onrampSessionId` didn't change and still processing. Otherwise, proceed as normal.